sunpy/database/tests/test_database.py

```diff
@@ -785,6 +787,40 @@ def test_fetch(database, download_query, tmpdir):
 
 
 @pytest.mark.online
+def test_fetch_separate_filenames():
+    # Setup
+    db = Database('sqlite:///')
+
+    download_query = [
+        vso.attrs.Time('2012-08-05', '2012-08-05 00:00:05'),
+        vso.attrs.Instrument('AIA')
+    ]
+
+    tmp_test_dir = os.path.join(
+        sunpy.config.get('downloads', 'download_dir'),
+        'tmp_test_dir/'
+    )
+
+    if not os.path.isdir(tmp_test_dir):
+        os.mkdir(tmp_test_dir)
+
+    path = tmp_test_dir + '{file}.fits'
+
+    db.fetch(*download_query, path=path)
+    
+    # Test
+    assert len(db) == 2
+    
+    dir_contents = os.listdir(tmp_test_dir)
+    assert dir_contents[-2] == 'aia_lev1_335a_2012_08_05t00_00_02_62z_image_lev1_fits.fits'
+    assert dir_contents[-1] == 'aia_lev1_94a_2012_08_05t00_00_01_12z_image_lev1_fits.fits'
+    assert dir_contents[-1].split('.')[0] != dir_contents[-2].split('.')[1]
```

Comment:
Though in my machine the test runs successfully, it doesn't in Travis... it seems that the order of the `dir_contents` it doesn't match... and [https://travis-ci.org/sunpy/sunpy/jobs/54440291#L1278](therefore compares 94 with 335). Could it be that `os.listdir` does not provide always the same order?  I've just made a test in my machine and I get this:
```
 'efz20000913_063611',
 'efz20000912_212410',
 'efz20000913_170011',
```
which it doesn't follow any logical order.  Would it better to use some regex? and see whether the file is there? or.. just check whether it exists:
```python
assert os.path.is_file(os.path.join(tmp_test_dir, 'aia_lev1_335a_2012_08_05t00_00_02_62z_image_lev1_fits.fits'))
```